src/ListView.js

```diff
@@ -0,0 +1,311 @@
+
+
+function ListView(props) {
+
+    // state data to be used later
+    const [selected, setSelected] = useState([]);
+    const [edited, setEdited] = useState(-1);
+    const [hidden, setHidden] = useState(false);
+    const [showWarning, setShowWarning] = useState(false);
+    const [showPriorities, setShowPriorities] = useState(false);
+    const [sort, setSort] = useState("creationAsc");
+    const [sortOptions, setSortOptions] = useState(false);
+    let showDeleteButton = selected.length > 0;
+    let disableChecks = (edited !== -1);
+
+    const qCreationAsc = query(collection(props.db, props.collection, props.listId, "Notes"));
+    const qCreationDesc = query(collection(props.db, props.collection, props.listId, "Notes"), orderBy("created", "desc"));
+    const qTextAsc = query(collection(props.db, props.collection, props.listId, "Notes"), orderBy("text", "asc"));
+    const qTextDesc = query(collection(props.db, props.collection, props.listId, "Notes"), orderBy("text", "desc"));
+    const qPriorityAsc = query(collection(props.db, props.collection, props.listId, "Notes"), orderBy("priority", "asc"));
+    const qPriorityDesc = query(collection(props.db, props.collection, props.listId, "Notes"), orderBy("priority", "desc"));
+
+    console.log(selected)
+    console.log(edited)
+    function collectionSelector() {
+        if (sort === "creationDesc") {
+            return (qCreationDesc)
+        } else if (sort === "textAsc") {
+            return (qTextAsc)
+        } else if (sort === "textDesc") {
+            return (qTextDesc)
+        } else if (sort === "priorityAsc") {
+            return (qPriorityAsc)
+        } else if (sort === "priorityDesc") {
+            return (qPriorityDesc)
+        } else{
+            return(qCreationAsc)
+        }
+    }
+    let [list,loading,error] = useCollectionData(collectionSelector());
+
+    if (error) {
+        console.log("ERROR: List data failed to load from Firestore")
+    }
+
+    let showHideButton = false;
+    let showSortButton = false;
+
+
+    if (!loading) {
+        showHideButton = list.filter(p => p.checked).length > 0;
+        showSortButton = list.length > 1;
+    }
+
+    function handleLineEdited(lineID) {
+        // if we're editing the bottommost 'Tap to Add Note' line and the text has changed, update our data/state
+        if (lineID !== -2 && document.getElementById('tempTapLine').value!=="") {
+                handleItemAdded2(document.getElementById('tempTapLine').value);
+            document.getElementById('tempTapLine').value=""
+        }
+        setEdited(lineID) // update edited line state
+    }
+
+    // changes line data for textboxes, checkboxes, or special key presses
+    function handleItemChanged(itemID, field, newValue) {
+        if (["text", "check_visible", "select_visible", "checked", "priority", "created"].includes(field)) {
+            updateDoc(doc(props.db, props.collection, props.listId, "Notes", itemID),
```

Comment:
Instead of .then(...), you can just use void updateDoc(...).

That'll also get rid of the warnings.